File: src/ios/WhatsAppSticker.m
/********* WhatsAppSticker.h Cordova Plugin Header *******/

#import <Cordova/CDVPlugin.h>

@interface WhatsAppSticker : CDVPlugin

- (void)sendToWhatsapp:(CDVInvokedUrlCommand*)command;

@end

/********* WhatsAppSticker.m Cordova Plugin Implementation *******/

#import <Cordova/CDVPlugin.h>

@implementation WhatsAppSticker

- (void)sendToWhatsapp:(CDVInvokedUrlCommand*)command
{

    // Obtém o JSON contendo as informações do Sticker Pack
    NSString* json = [command.arguments objectAtIndex:0];

    // Copia as informações do JSON para o Pasteboard, que é a forma na qual o Whatsapp lê as informações do Sticker Pack no iOS
    [[UIPasteboard generalPasteboard] setData:json forPasteboardType:@"net.whatsapp.third-party.sticker-pack"];

    UIApplication *application = [UIApplication sharedApplication];

    // O URL Scheme usado para enviar as informações do Sticker Pack
    NSURL *URL = [NSURL URLWithString:@"whatsapp://stickerPack"];

    // Abre o URL Scheme, e por consequencia, abre o WhatsApp
    [application openURL:URL options:@{} completionHandler:^(BOOL success) {
        CDVPluginResult* pluginResult = nil;
        if (success) {
            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:@"Figurinhas enviadas."];
        } else {
            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"Ocorreu algum problema, as figurinhas não foram enviadas."];
        }

        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    }];
}

@end

File: src/ios/WhatsAppSticker.swift
import UIKit

@objc(WhatsAppSticker) class WhatsAppSticker : CDVPlugin {

  private let PasteboardExpirationSeconds: TimeInterval = 60
  private let PasteboardStickerPackDataType: String = "net.whatsapp.third-party.sticker-pack"
  private let WhatsAppURL: URL = URL(string: "whatsapp://stickerPack")!
    
  func sendToWhatsapp(_ command: CDVInvokedUrlCommand) {
    var pluginResult = CDVPluginResult(
      status: CDVCommandStatus_ERROR
    )

    let pasteboard: UIPasteboard = UIPasteboard.general
    let json = command.arguments[0] as? String ?? ""
    let jsonData = Data(json.utf8);

    if #available(iOS 10.0, *) {
        pasteboard.setItems([[PasteboardStickerPackDataType: jsonData]], options: [UIPasteboardOption.localOnly: true, UIPasteboardOption.expirationDate: NSDate(timeIntervalSinceNow: PasteboardExpirationSeconds)])
    } else {
        pasteboard.setData(jsonData, forPasteboardType: PasteboardStickerPackDataType)
    }
    
    DispatchQueue.main.async {
      if UIApplication.shared.canOpenURL(URL(string: "whatsapp://stickerPack")!) {
          if #available(iOS 10.0, *) {
              UIApplication.shared.open(self.WhatsAppURL, options: [:], completionHandler: nil)
          } else {
              UIApplication.shared.openURL(self.WhatsAppURL)
          }
      }
    }

    self.commandDelegate!.send(
      pluginResult, 
      callbackId: command.callbackId
    )
  }
}